src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -3108,6 +3115,28 @@ int Pane::GetLeafPaneCount() const noexcept
     return _IsLeaf() ? 1 : (_firstChild->GetLeafPaneCount() + _secondChild->GetLeafPaneCount());
 }
 
+// Method Description:
+// - Should be called when this pane is created via a default terminal handoff
+// - Finalizes our configuration given the information that we have been
+//   created via default handoff
+void Pane::FinalizeConfigurationGivenDefault()
+{
+    if (_IsLeaf())
+    {
+        if (_profile)
+        {
+            if (_profile.CloseOnExit() == CloseOnExitMode::GracefulIfLaunchedByTerminal)
+            {
+                // We only want to close 'gracefully' if we were launched by Terminal
+                // Since we were launched via defterm, override this (i.e. it will
+                // be treated as 'always', see _ControlConnectionStateChangedHandler
+                // for where this boolean is used, and see GH #13325 for discussion)
+                _overrideCloseOnExit = true;
+            }
+        }
+    }
+}

```

Comment:
```suggestion
    if (_IsLeaf() && _profile && _profile.CloseOnExit() == CloseOnExitMode::GracefulIfLaunchedByTerminal)
    {
        // We only want to close 'gracefully' if we were launched by Terminal
        // Since we were launched via defterm, override this (i.e. it will
        // be treated as 'always', see _ControlConnectionStateChangedHandler
        // for where this boolean is used, and see GH #13325 for discussion)
        _overrideCloseOnExit = true;
    }
```

or at least you can combine the `_profile` and close on exit check to remove a level of nesting

Reply:
Combined, thanks!